src/components/SelectComponent.tsx

```diff
@@ -40,14 +40,17 @@ export class SelectComponent extends React.Component<InlineComponentProps, any>
         }
         let foreignTitle = !hideLabel ? "Choose " + this.props.field.title : "Choose"
         if (this.props.field.foreign) {
-            if (!this.props.field.foreign.key) {
-                console.error("Did you forget to add a \"key\" field in foreign . Possible culprit: ", this.props.field)
+            if (!this.props.field.foreign.key && !this.props.field.foreign.keys) {
+                console.error("Did you forget to add a \"key(s)\" field in foreign . Possible culprit: ", this.props.field)
             }
             if (!this.props.field.foreign.title) {
                 console.error("Did you forget to add a \"title\" field in foreign . Possible culprit: ", this.props.field)
             }
             if (!_.isEmpty(this.props.currentModel)) {
                 const foreignDoc = _.find(optionsData, (doc: any) => {
+                    if (this.props.field.foreign.keys) {
```

Comment:
should this block be after the valueType === "object" block. I am not sure why I added the valueType thing. Can you do a code search to see if anyone is using it. If not, we can remove that, seems hacky

Reply:
Also we should check if both key and keys are passed ... we can do a console.error in such a case

Reply:
problem with `valueType === "object"` condition is that its relying on `id` field, wanted to keep this backward compatible, so checking for `keys` first.

`valueType === object` might make sense with `transform` if we want to refer to some customization of foreign object.

Reply:
will add console.error if both passed